### PR TITLE
[FIX] payment: do not add fee for validation transaction

### PR DESCRIPTION
**Pull request by hote-odoo**
When making transaction with 'validation' type and where fees
were active, fees were added to the amount  of the transaction. 
This behavior is problematic because
validation operations are used for tokenization for example where
no credit should be charged to the customer.
Now validation operations have no fees even if fees are active
on the acquirer.

Task - 2784765

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by hote-odoo** [comment]
There is a modification on the stripe acquirer for testing purposes, before merging the fees should be removed from stripe.

File: addons/payment/models/payment_acquirer.py [hunk 1]
```diff
@@ -350,7 +350,7 @@ def _should_build_inline_form(self, is_validation=False):
         """
         return True
 
-    def _compute_fees(self, amount, currency, country):
+    def _compute_fees(self, amount, currency, country, operation=None):
```
Comment (Feyensv) [inline addons/payment/models/payment_acquirer.py:353]: If there is only one place where the operation is given, You should only manage it there (in the tx creation), without modifying this method.

File: addons/payment_stripe/data/payment_acquirer_data.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <record id="payment.payment_acquirer_stripe" model="payment.acquirer">
         <field name="provider">stripe</field>
         <field name="support_authorization">False</field>
-        <field name="support_fees_computation">False</field>
+        <field name="support_fees_computation">True</field>
```
Comment (Feyensv) [inline addons/payment_stripe/data/payment_acquirer_data.xml:7]: Ideally, temporary test modifications should be in a separate commit with a clear title saying the commit shouldn't be merged ;).

Comment (hote-odoo) [reply addons/payment_stripe/data/payment_acquirer_data.xml:7]: Thank you :)

File: addons/payment/models/payment_acquirer.py (hunk 1 above)
Comment (hote-odoo) [reply addons/payment/models/payment_acquirer.py]: fixed it

File: addons/payment/models/payment_transaction.py [hunk 3]
```diff
@@ -193,11 +193,14 @@ def create(self, values_list):
...
-                values.get('amount', 0), currency, partner.country_id
-            )
+            # Compute fees, for validation transactions fees are zero
+            if values.get('operation') == 'validation':
+                values['fees'] = 0
+            else:
+                currency = self.env['res.currency'].browse(values.get('currency_id')).exists()
+                values['fees'] = acquirer._compute_fees(
+                    values.get('amount', 0), currency, partner.country_id,
+                )
```
Comment (AntoineVDV) [inline addons/payment/models/payment_transaction.py:203]:
```suggestion
            # Compute fees
            if values.get('operation') != 'validation':
                currency = self.env['res.currency'].browse(values.get('currency_id')).exists()
                values['fees'] = acquirer._compute_fees(
                    values.get('amount', 0.), currency, partner.country_id
                )
            else:  # No fees are charged for validation transactions.
                values['fees'] = 0.
```

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+
